File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/digitaltwin/GetObjectElementListResponse.java
package org.opcfoundation.webserver.types.message.digitaltwin;

import org.opcfoundation.webserver.types.ObjectElementDescriptor;
import org.opcfoundation.webserver.types.ReferenceTargetDescriptor;

import java.util.ArrayList;
import java.util.List;

public class GetObjectElementListResponse {
    private final List<ObjectElementDescriptor> elements;
    private boolean containsMoreData;

    public GetObjectElementListResponse()
    {
        this.elements = new ArrayList<>();
        this.containsMoreData = false;
    }

    public List<ObjectElementDescriptor> getElements() {
        return elements;
    }

    public void add(ObjectElementDescriptor descriptor)
    {
        elements.add(descriptor);
    }

    public boolean containsMoreData()
    {
        return containsMoreData;
    }

    public void setContainsMoreData(boolean containsMoreData) {
        this.containsMoreData = containsMoreData;
    }
}

File: opcua-webservice-node/source/src/server/service/message/digitialtwin/GetObjectElementListResponse.ts
import { ObjectElementDescriptor } from "../../../types/digitaltwin/ObjectElementDescriptor";

export class GetObjectElementListResponse {
    private readonly _elements: Array<ObjectElementDescriptor>;
    private _containsMoreData: boolean;

    constructor() {
        this._elements = [];
        this._containsMoreData = false;
    }

    get elements(): Array<ObjectElementDescriptor> {
        return this._elements;
    }

    add(descriptor: ObjectElementDescriptor): void {
        this._elements.push(descriptor);
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }

    set containsMoreData(containsMoreData: boolean) {
        this._containsMoreData = containsMoreData;
    }
}